distsql/stream.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package distsql
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tipb/go-tipb"
+	goctx "golang.org/x/net/context"
+)
+
+// streamResult implements the SelectResult interface.
+type streamResult struct {
+	resp       kv.Response
+	rowLen     int
+	fieldTypes []*types.FieldType
+	ctx        context.Context
+
+	// NOTE: chunk == nil means finish, while len(chunk.RowsData) == 0 doesn't.
+	chunk *tipb.Chunk
```

Comment:
Can we rename it? rawData、copData or others

Reply:
No, it's in protobuf....
One day we'll get rid of it, maybe.

Reply:
I mean the field of `chunk`.